vcd/datasource_vcd_resource_list.go

```diff
@@ -0,0 +1,720 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+type resourceRef struct {
+	name string
+	id   string
+	href string
+}
+
+func datasourceVcdResourceList() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdResourceListRead,
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			// Parent will be needed for:
+			// * VM (parent: vApp)
+			// * catalogItem (catalog)
+			// * mediaItem (catalog)
+			// * all edge gateway objects (NAT, firewall, lb)
+			// When the parent is org or vdc, they are taken from the regular fields above
+			"parent": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the parent to the resources being retrieved",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Unique name of the Info",
+			},
+			"resource_type": &schema.Schema{
```

Comment:
maybe worth it to list possible values in description and validate them

Reply:
added